From c72abfee36efbb60d90f13abb6c29c1ea11f5024 Mon Sep 17 00:00:00 2001
From: Atul Bajaj <atul.bajaj@intel.com>
Date: Tue, 30 Jan 2018 10:36:37 +0530
Subject: [PATCH 1/4] Protex issues OAM-56266: Initialized array space and
 removed unused sink variable to resolve Protex issue.

Change-Id: Iee2d69665f102cca7f34fdf498590cb6486d0389
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-56266
Signed-off-by: Atul Bajaj <atul.bajaj@intel.com>
---
 runtime/thread.cc | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/runtime/thread.cc b/runtime/thread.cc
index cdbb908..f86bd71 100644
--- a/runtime/thread.cc
+++ b/runtime/thread.cc
@@ -580,8 +580,9 @@ void Thread::InstallImplicitProtection() {
 #else
           1u;
 #endif
-      volatile char space[kPageSize - (kAsanMultiplier * 256)];
-      char sink ATTRIBUTE_UNUSED = space[zero];
+      //OAM-56266: Initialized array space and removed unused sink variable
+      volatile char space[kPageSize - (kAsanMultiplier * 256)]={0};
+      //char sink ATTRIBUTE_UNUSED = space[zero];
       if (reinterpret_cast<uintptr_t>(space) >= target + kPageSize) {
         Touch(target);
       }
-- 
1.9.1

